### ISSUE TITLE
Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'launch path not accessible'

### ISSUE THREAD

**Issue by brooksc** [WPO-Foundation/xrecord#8]
Thanks for creating this -- just tried running it and it fails though, any ideas?

Using MacOS Sierra 10.12.6 (16G1114)

using binary from bin folder and received the error below.  "-h" does show help output, but -l or --list fail.  Xcode is installed and update to date.  

> xrecord -l
2018-02-08 14:48:58.010 xrecord[23911:21927265] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'launch path not accessible'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fffbc0972cb __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fffd0eae48d objc_exception_throw + 48
	2   CoreFoundation                      0x00007fffbc115c3d +[NSException raise:format:] + 205
	3   Foundation                          0x00007fffbdabfa0e -[NSConcreteTask launchWithDictionary:] + 414
	4   xrecord                             0x000000010d204164 xrecord + 2601316
	5   libdyld.dylib                       0x00007fffd1794235 start + 1
	6   ???                                 0x0000000000000002 0x0 + 2
)
libc++abi.dylib: terminating with uncaught exception of type NSException

**Comment by gnissyan** [comment]
The same issue with almost all arguments, one way to avoid that exception is to set also **-d**

**Comment by brooksc** [comment]
I actually had it working briefly with -d option!  but now it just throws a segmentation fault...

> xrecord -d -q -i cb60fe0607cda908b399c13f429edf5aa3e67fa3 -o foo.mov
2018-07-19 11:19:25.433 xrecord[5611:278703] Starting capture....
Recording started.  Hit ctrl-C to stop.
2018-07-19 11:19:28.931 xrecord[5611:278703] Recording started.  Hit ctrl-C to stop.
^CStopping recording...
2018-07-19 11:19:36.421 xrecord[5611:278703] Stopping recording...
2018-07-19 11:19:36.421 xrecord[5611:278703] captureOutput Finished callback
Done
2018-07-19 11:19:36.431 xrecord[5611:278703] Done
zsh: segmentation fault  xrecord -d -q -i cb60fe0607cda908b399c13f429edf5aa3e67fa3 -o foo.mov

**Comment by gnissyan** [comment]
Have you tried to do the device screen recording with the QuickTime Player only? Does it work smoothly?

**Comment by brooksc** [comment]
Yes, perfectly fine.  I finally upgraded to High Sierra 10.13.6 and xrecord -l is still crashing

```$ xrecord -l
2018-09-19 14:24:45.985 xrecord[32827:4255224] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'launch path not accessible'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff2c64e2db __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fff537ebc76 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff2c6dfd7d +[NSException raise:format:] + 205
	3   Foundation                          0x00007fff2e8af992 -[NSConcreteTask launchWithDictionary:error:] + 1304
	4   xrecord                             0x00000001020d3164 xrecord + 2601316
	5   libdyld.dylib                       0x00007fff54405015 start + 1
	6   ???                                 0x0000000000000002 0x0 + 2
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```
